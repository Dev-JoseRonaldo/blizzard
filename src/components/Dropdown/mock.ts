import { ButtonFooterDropdownProps } from '.'

export const FooterDropdownGamesMock = [
  {
    id: '0',
    text: 'Ver todos jogos',
    iconUrl:
      'https://res.cloudinary.com/devjoseronaldo/image/upload/v1672528852/blizzard/icons/icon-footer-dropdown-1_esmtd5.png',
    alt: 'Alternative Text',
    width: 9,
    height: 9,
  },
  {
    id: '1',
    text: 'Aplicativo Battle.net',
    iconUrl:
      'https://res.cloudinary.com/devjoseronaldo/image/upload/v1672528852/blizzard/icons/icon-footer-dropdown-2_zlfi4r.png',
    alt: 'Alternative Text',
    width: 24,
    height: 24,
  },
  {
    id: '2',
    text: 'Downloads',
    iconUrl:
      'https://res.cloudinary.com/devjoseronaldo/image/upload/v1672528852/blizzard/icons/icon-footer-dropdown-3_qyrikv.png',
    alt: 'Alternative Text',
    width: 24,
    height: 24,
  },
  {
    id: '3',
    text: 'FÃ³runs dos jogos',
    iconUrl:
      'https://res.cloudinary.com/devjoseronaldo/image/upload/v1672528852/blizzard/icons/icon-footer-dropdown-4_oi11bo.png',
    alt: 'Alternative Text',
    width: 24,
    height: 24,
  },
] as ButtonFooterDropdownProps[]

export const FooterDropdownSportsMock = [
  {
    id: '0',
    text: 'Torneios da comunidade',
    iconUrl:
      'https://res.cloudinary.com/devjoseronaldo/image/upload/v1672528852/blizzard/icons/icon-footer-dropdown-5_s01gk9.png',
    alt: 'Alternative Text',
    width: 24,
    height: 23,
  },
] as ButtonFooterDropdownProps[]
