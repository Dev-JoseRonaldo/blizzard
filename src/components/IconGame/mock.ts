import { IconGameProps } from '.'

export const IconGamesMock = [
  {
    id: '0',
    iconUrl:
      'https://res.cloudinary.com/devjoseronaldo/image/upload/v1672246110/blizzard/games/icon-game6-sm_gyr5x3.png',
    alt: 'Diablo 2 ressurected',
    width: 70,
    height: 70,
    name: 'Diablo® II ressurected',
  },
  {
    id: '1',
    iconUrl:
      'https://res.cloudinary.com/devjoseronaldo/image/upload/v1672508838/blizzard/games/icon-game12_bsm8fg.png',
    alt: 'Overwatch',
    width: 70,
    height: 70,
    name: 'Overwatch®',
  },
  {
    id: '2',
    iconUrl:
      'https://res.cloudinary.com/devjoseronaldo/image/upload/v1672246107/blizzard/games/icon-game7-sm_ac8sa0.png',
    alt: 'Overwatch 2',
    width: 70,
    height: 70,
    name: 'Overwatch® 2',
  },
  {
    id: '3',
    iconUrl:
      'https://res.cloudinary.com/devjoseronaldo/image/upload/v1672246098/blizzard/games/icon-game3-sm_ohqazb.png',
    alt: 'World of Warcraft',
    width: 70,
    height: 70,
    name: 'World of Warcraft®',
  },
  {
    id: '4',
    iconUrl:
      'https://res.cloudinary.com/devjoseronaldo/image/upload/v1672246098/blizzard/games/icon-game2-sm_ugeylz.png',
    alt: 'Hearthstone',
    width: 70,
    height: 70,
    name: 'Hearthstone®',
  },
  {
    id: '5',
    iconUrl:
      'https://res.cloudinary.com/devjoseronaldo/image/upload/v1672246110/blizzard/games/icon-game8-sm_doljts.png',
    alt: 'Heroes of the storm',
    width: 70,
    height: 70,
    name: 'Heroes of the storm®',
  },
  {
    id: '6',
    iconUrl:
      'https://res.cloudinary.com/devjoseronaldo/image/upload/v1672246108/blizzard/games/icon-game9-sm_czfq5r.png',
    alt: 'Warcraft 3 Reforged',
    width: 70,
    height: 70,
    name: 'Warcraft® III Reforged',
  },
  {
    id: '7',
    iconUrl:
      'https://res.cloudinary.com/devjoseronaldo/image/upload/v1672246098/blizzard/games/icon-game1-sm_guifbg.png',
    alt: 'Diablo IV',
    width: 70,
    height: 70,
    name: 'Diablo® IV',
  },
  {
    id: '8',
    iconUrl:
      'https://res.cloudinary.com/devjoseronaldo/image/upload/v1672246115/blizzard/games/icon-game4-sm_omatvu.png',
    alt: 'Diablo Immortal',
    width: 70,
    height: 70,
    name: 'Diablo® Immortal',
  },
  {
    id: '9',
    iconUrl:
      'https://res.cloudinary.com/devjoseronaldo/image/upload/v1672246107/blizzard/games/icon-game10-sm_ur7sg4.png',
    alt: 'Diablo III',
    width: 70,
    height: 70,
    name: 'Diablo® III',
  },
  {
    id: '10',
    iconUrl:
      'https://res.cloudinary.com/devjoseronaldo/image/upload/v1672246107/blizzard/games/icon-game5-sm_x8mv3o.png',
    alt: 'StarCraft II',
    width: 70,
    height: 70,
    name: 'StarCraft® II',
  },
  {
    id: '11',
    iconUrl:
      'https://res.cloudinary.com/devjoseronaldo/image/upload/v1672246106/blizzard/games/icon-game11-sm_jh6ppw.png',
    alt: 'StarCraft Remastered',
    width: 70,
    height: 70,
    name: 'StarCraft® Remastered',
  },
] as IconGameProps[]

export const IconSportsMock = [
  {
    id: '0',
    iconUrl:
      'https://res.cloudinary.com/devjoseronaldo/image/upload/v1672246117/blizzard/games/sport1-lg_zjwsp4.png',
    alt: 'Hearthstone masters',
    width: 174,
    height: 149,
    name: 'Hearthstone® masters',
  },
  {
    id: '1',
    iconUrl:
      'https://res.cloudinary.com/devjoseronaldo/image/upload/v1672246108/blizzard/games/sport2-lg_kclg6q.png',
    alt: 'Campeonato Mundial de Arena WoW',
    width: 174,
    height: 149,
    name: 'Campeonato Mundial de Arena WoW®',
  },
  {
    id: '2',
    iconUrl:
      'https://res.cloudinary.com/devjoseronaldo/image/upload/v1672246112/blizzard/games/sport3-lg_fiendi.png',
    alt: 'StarCraft 2 WCS',
    width: 174,
    height: 149,
    name: 'StarCraft® II WCS',
  },
  {
    id: '3',
    iconUrl:
      'https://res.cloudinary.com/devjoseronaldo/image/upload/v1672246112/blizzard/games/sport4-lg_yv96gb.png',
    alt: 'Copa Mundial de Overwatch',
    width: 174,
    height: 149,
    name: 'Copa Mundial de Overwatch®',
  },
  {
    id: '4',
    iconUrl:
      'https://res.cloudinary.com/devjoseronaldo/image/upload/v1672246116/blizzard/games/sport5-lg_zzzbjx.png',
    alt: 'Liga de Overwatch',
    width: 174,
    height: 149,
    name: 'Liga de Overwatch®',
  },
] as IconGameProps[]
