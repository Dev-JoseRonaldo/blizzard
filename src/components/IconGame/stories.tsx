import { ComponentMeta, ComponentStory } from '@storybook/react'
import { IconGame } from '.'

export default {
  title: 'Components/IconGame',
  component: IconGame,
  parameters: {
    backgrounds: {
      default: 'dark',
    },
    design: {
      type: 'figma',
      url: 'FIGMA_URL_FRAME',
    },
  },
  args: {
    iconUrl:
      'https://res.cloudinary.com/devjoseronaldo/image/upload/v1672246117/blizzard/games/sport1-lg_zjwsp4.png',
    alt: 'Texto alternativo',
    width: 174,
    height: 149,
  },
} as ComponentMeta<typeof IconGame>

const Template: ComponentStory<typeof IconGame> = (args) => (
  <IconGame {...args} />
)

export const Default = Template.bind({})
Default.args = {}

export const WithText = Template.bind({})
WithText.args = {
  name: 'HearthstoneÂ® masters',
}
