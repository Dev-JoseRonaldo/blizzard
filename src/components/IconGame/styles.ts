export const ContainerStyle = `group flex flex-col items-center justify-center gap-[0.5375rem]`

export const ImageStyle = `group-hover:translate-y-[-0.2rem] duration-300 transition-all`

export const TextStyle = `text-gray-600 font-medium font-secondary text-sm max-w-[7rem] text-center group-hover:text-gray-500 duration-100 transition-all ease-in`
