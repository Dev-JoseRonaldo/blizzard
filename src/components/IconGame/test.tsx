import { render, screen, waitFor } from '@testing-library/react'

import { IconGame } from '.'

const { getByRole, getByText, queryByText } = screen

const data = {
  iconUrl: 'https://res.cloudinary.com/icon.png',
  alt: 'Alternative Text',
  width: 174,
  height: 149,
}

describe('<IconGame />', () => {
  it('should render by default', async () => {
    render(<IconGame {...data} />)

    const name = queryByText('name')
    const icon = getByRole('img', { name: 'Alternative Text' })

    expect(name).not.toBeInTheDocument()
    expect(icon).toBeInTheDocument()

    await waitFor(() => {
      expect(icon).toHaveAttribute(
        'src',
        '/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Ficon.png&w=384&q=75',
      )
    })

    expect(icon).toHaveAttribute('alt', 'Alternative Text')
    expect(icon).toHaveStyle({
      width: 174,
      height: 149,
    })
  })

  it('should render with text', async () => {
    render(<IconGame {...data} name="name" />)

    const name = getByText('name')

    expect(name).toBeInTheDocument()
  })

  it('Should match snapshot', () => {
    const { container } = render(<IconGame {...data} />)

    expect(container).toMatchSnapshot()
  })
})
