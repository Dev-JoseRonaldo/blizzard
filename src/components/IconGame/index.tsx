import Image from 'next/image'

import { ContainerStyle, ImageStyle, TextStyle } from './styles'

export interface IconGameProps {
  iconUrl: string
  alt: string
  width: number
  height: number
  name?: string
}

export const IconGame = ({
  iconUrl,
  alt,
  width,
  height,
  name,
}: IconGameProps) => {
  return (
    <figure className={`${ContainerStyle} ${name && 'cursor-pointer'}`}>
      <Image
        src={iconUrl}
        alt={alt}
        width={width}
        height={height}
        className={name ? ImageStyle : ''}
      />
      {name && <figcaption className={TextStyle}>{name}</figcaption>}
    </figure>
  )
}
