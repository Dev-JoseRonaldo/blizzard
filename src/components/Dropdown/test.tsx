import { render, screen, fireEvent } from '@testing-library/react'

import { Dropdown, DropdownProps } from '.'
import { IconSportsMock } from '../IconGame/mock'
import { FooterDropdownSportsMock } from './mock'

const { getByRole, queryByRole, getByText, queryByText } = screen

const data: DropdownProps = {
  items: IconSportsMock,
  type: 'sports',
  name: 'sports',
  FooterItems: FooterDropdownSportsMock,
}

describe('<Dropdown />', () => {
  it('should render by default', () => {
    render(<Dropdown {...data} />)

    const dropdownButton = getByRole('button', { name: 'sports' })
    const imgItem = queryByRole('img', { name: data.items[0].alt })
    const footerTextItem = queryByText(data.FooterItems[0].text)

    expect(dropdownButton).toBeInTheDocument()
    expect(footerTextItem).not.toBeInTheDocument()
    expect(imgItem).not.toBeInTheDocument()
  })

  it('should render Dropdown content', async () => {
    render(<Dropdown {...data} />)

    const dropdownButton = getByRole('button', { name: 'sports' })

    fireEvent.click(dropdownButton)

    for (let i = 0; i < data.items.length; i++) {
      const imgItem = queryByRole('img', { name: data.items[i].alt })

      expect(imgItem).toBeInTheDocument()
      expect(imgItem).toHaveAttribute('alt', data.items[i].alt)
    }

    for (let i = 0; i < data.FooterItems.length; i++) {
      const footerTextItem = getByText(data.FooterItems[i].text)

      expect(footerTextItem).toBeInTheDocument()
    }
  })

  it('should match snapshot', () => {
    const { container } = render(<Dropdown {...data} />)

    expect(container).toMatchSnapshot()
  })
})
