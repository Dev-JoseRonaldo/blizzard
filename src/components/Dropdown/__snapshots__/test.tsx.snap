// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> should match snapshot 1`] = `
<div>
  <button
    aria-expanded="false"
    aria-haspopup="menu"
    class="flex items-center justify-center gap-3.5 top-0 z-10 text-white rounded"
    data-headlessui-state=""
    id="headlessui-menu-button-:r8:"
    type="button"
  >
    sports
    <svg
      class="transition-all rotate-0"
      fill="white"
      font-size="24"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g>
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 13.172l4.95-4.95 1.414 1.414L12 16 5.636 9.636 7.05 8.222z"
        />
      </g>
    </svg>
  </button>
</div>
`;
